vtk_viewer/src/vtk_utils.cpp

```diff
@@ -410,6 +410,13 @@ void generateNormals(vtkSmartPointer<vtkPolyData>& data, int flip_normals)
         norm[1] /= pts->GetNumberOfIds();
         norm[2] /= pts->GetNumberOfIds();
 
+        double mag = std::sqrt(norm[0] * norm[0] + norm[1] * norm[1] + norm[2] * norm[2]);
+
+        // Normalize vector
+        norm[0] /= mag;
+        norm[1] /= mag;
+        norm[2] /= mag;
+
         // set the normal for the given cell
```

Comment:
I have no real issue. There's no error reporting mechanism atm, so it may not matter to much but can we check if the computed magnitude is near zero?